0017-negotiation-and-execution-protocol.adoc

```diff
@@ -0,0 +1,451 @@
+= Separating negotiation and execution phase =
+Thomas Eizinger <thomas@coblox.tech>; Philipp Hoenisch <philipp@coblox.tech>; Daniel Karzel <daniel.karzel@coblox.tech>;
+:toc:
+:revdate: 2019-10-05
+:sectnums:
+:sectnumlevels: 5
+
+NOTE: Author: {authors} +
+Date: {revdate} +
+Tracking issue: https://github.com/comit-network/comit-rs/issues/1413[#1413]
+
+== Context ==
+
+Over time we came to the conclusion that our protocol stack in general, and RFC003 in particular, is difficult to understand in certain ways.
+
+We could specify several problems:
+
+. The Accept/Decline _feels weird_. It often does not make sense to _accept_ a trade which was already agreed on before (e.g. on application layer).
+. We do not clearly distinguish rate negotiation from the execution at the moment.
+. It is hard to add the concept of orders and an orderbook to COMIT.
+
+=== This spike includes
+
+. Solution proposal for splitting trade negotiation and execution in different protocols.
+. Solution proposal for the "How to update an order"/"Orders should not get stale" problem.
+. Defining the base for a first negotiation protocol.
+. Defining the changes needed to the existing execution protocol RFC003.
+
+=== This spike does not include
+
+The needed API changes to facilitate the newly proposed execution protocol.
+
+== TL;DR & Next step proposal
+
+This spike proposes a new protocol to exchange trade orders between a maker and a taker including an update to the execution protocol as defined in RFC003.
+
+We propose the following steps to achieve early validations of the proposed protocol (note: this proposal assumes that none of the assumptions have been refuted and does not make any assumptions on the size of the tickets or PRs):
+
+. Integrate into the https://github.com/comit-network/create-comit-app/[create-commit-app] a simple request/response example to exchange of order parameters (KISS: keep it simple stupid).
```

Comment:
So totally discard the diagrams and just do something that works? Why?

Reply:
What? No. Sorry if this was not clear. KISS was meant in regards of the communication protocol, i.e. don't go for libp2p or anything for now, just do a simple request/response pattern maybe using HTTP.